server/api/embyconnect.ts

```diff
@@ -0,0 +1,207 @@
+    const linkedServers = await this.getValidServers(
+      connectAuthResponse.User.Id,
+      connectAuthResponse.AccessToken
+    );
+
+    const matchingServer = this.findMatchingServer(linkedServers);
+
+    const embyServerApi = new EmbyServerApi(getHostname(), this.ClientIP);
+    const localUserExchangeResponse = await embyServerApi.localAuthExchange(
+      matchingServer.AccessKey,
+      connectAuthResponse.User.Id,
+      this.DeviceId
+    );
+
+    return {
+      User: {
+        Name: connectAuthResponse.User.Name,
+        ServerId: matchingServer.SystemId,
+        ServerName: matchingServer.Name,
+        Id: localUserExchangeResponse.LocalUserId,
+        Configuration: {
+          GroupedFolders: [],
+        },
+        Policy: {
+          IsAdministrator: false, // This requires an additional EmbyServer API call, skipping for now
+        },
+      },
+      AccessToken: localUserExchangeResponse.AccessToken,
+    } as JellyfinLoginResponse;
+  }
+
+  private async getConnectUserAccessToken(
+    Email?: string,
+    Password?: string
+  ): Promise<ConnectAuthResponse> {
+    try {
+      const textResponse = await this.post<string>(
+        '/service/user/authenticate',
+        { nameOrEmail: Email, rawpw: Password },
+        {},
+        undefined,
+        {
+          headers: {
+            'Content-Type': 'application/x-www-form-urlencoded',
+          },
+        }
+      );
+
+      return JSON.parse(textResponse) as ConnectAuthResponse;
+    } catch (e) {
+      logger.debug(`Failed to authenticate using EmbyConnect: ${e.message}`, {
+        label: 'EmbyConnect API',
+        ip: this.ClientIP,
+      });
+      throw new ApiError(e.cause?.status, ApiErrorCode.InvalidCredentials);
+    }
+  }
+
+  private async getValidServers(
+    ConnectUserId: string,
+    AccessToken: string
+  ): Promise<LinkedServer[]> {
+    try {
+      const textResponse = await this.get<string>(
+        `/service/servers`,
+        { userId: ConnectUserId },
+        undefined,
+        {
+          headers: {
+            'X-Connect-UserToken': AccessToken,
+          },
+        }
+      );
+
+      return JSON.parse(textResponse) as LinkedServer[];
+    } catch (e) {
+      logger.error(
+        `Failed to retrieve EmbyConnect user server list: ${e.message}`,
+        {
+          label: 'EmbyConnect API',
+          ip: this.ClientIP,
+        }
+      );
+      throw new ApiError(e.cause?.status, ApiErrorCode.InvalidAuthToken);
+    }
+  }
+
+  private findMatchingServer(linkedEmbyServers: LinkedServer[]): LinkedServer {
+    const settings = getSettings();
+    const matchingServer = linkedEmbyServers.find(
+      (server) => server.SystemId === settings.jellyfin.serverId
+    );
+
+    if (!matchingServer) {
+      throw new Error(
+        `No matching linked Emby server found for serverId: ${settings.jellyfin.serverId}`
+      );
+    }
+
+    return matchingServer;
+  }
+}
+
+class EmbyServerApi extends ExternalAPI {
```

Comment:
Could you move the only method of this class to the EmbyConnectAPI class? I don't think it's a good idea to create a class just for one method.

Reply:
Done.

The reason for the additional class was to bypass the baseUrl already passed to the EmbyConnectAPI constructor, because this call requires the server host url and not the EmbyConnect address.

In order to satisfy the request to only have a method, I had to expose the `overwriteBaseUrl?: string` argument in the ExternalApi `get()` method. I don't believe this is an issue as it's already done in other methods.

Reply:
Ok I get it. In my opinion it makes more sense to not use this overwriteBaseUrl option. This option was introduced while transitionning from Axios to the Fetch API, for compatibility reason.
I think it makes things confusing to use this, since the `.get`, `.post`, etc methods should be used to GET/POST/etc data only on the same domain used everywhere in the class.
You could use the regular Fetch API instead, with something like this:
```js
const res = await fetch('...', { headers: { ... }});
if (!res.ok) throw new Error('...');
return await res.json();
```